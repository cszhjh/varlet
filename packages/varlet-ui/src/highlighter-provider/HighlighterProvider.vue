<script lang="ts">
import { defineComponent, h } from 'vue'
import { props } from './props'
import { createNamespace } from '../utils/components'
import { call } from '@varlet/shared'
import { provideHighlighterProvider } from './provide'

const { name, n } = createNamespace('highlighter-provider')

export default defineComponent({
  name,
  props,
  setup(props, { slots }) {
    provideHighlighterProvider({ highlighter: props.highlighter, theme: props.theme })

    return () =>
      h(
        props.tag,
        {
          class: n(),
        },
        call(slots.default)
      )
  },
})
</script>
