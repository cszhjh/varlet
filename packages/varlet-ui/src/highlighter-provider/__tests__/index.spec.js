import HighlighterProvider from '../index'
import VarHighlighterProvider from '../HighlighterProvider'
import { createApp } from 'vue'
import { mount } from '@vue/test-utils'
import { expect, test } from 'vitest'
import { codeToHtml } from 'shiki'

test('test highlighterProvider component plugin', () => {
  const app = createApp({}).use(HighlighterProvider)
  expect(app.component(HighlighterProvider.name)).toBeTruthy()
})

test('test highlighterProvider highlighter', () => {
  const wrapper = mount(VarHighlighterProvider, {
    props: {
      highlighter: {
        codeToHtml,
      },
    },
  })
  expect(wrapper.html()).toMatchSnapshot()

  wrapper.unmount()
})

test('test highlighterProvider tag', () => {
  const wrapper = mount(VarHighlighterProvider, { props: { tag: 'span' } })
  expect(wrapper.html()).toMatchSnapshot()

  wrapper.unmount()
})
