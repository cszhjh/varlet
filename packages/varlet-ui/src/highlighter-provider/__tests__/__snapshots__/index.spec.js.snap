// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test highlighterProvider highlighter 1`] = `"<div class="var-highlighter-provider"></div>"`;

exports[`test highlighterProvider tag 1`] = `"<span class="var-highlighter-provider"></span>"`;
